clients/js/src/createTree.ts

```diff
@@ -27,6 +27,18 @@ export const createTree = async (
     getMerkleTreeSize(input.maxDepth, input.maxBufferSize, input.canopyDepth);
   const lamports = await context.rpc.getRent(space);
 
+  let programId;
+  if (input.compressionProgram) {
+    programId = Array.isArray(input.compressionProgram)
+      ? input.compressionProgram[0]
+      : input.compressionProgram;
+  } else {
+    programId = context.programs.getPublicKey(
+      'splAccountCompression',
+      SPL_ACCOUNT_COMPRESSION_PROGRAM_ID
+    );
+  }
```

Comment:
Is the purpose of `isArray` to determine if `compressionProgram` is a `Pda` instead of a `PublicKey`? If so, we can also use the Umi `publicKey` function, which accepts `PublicKeyInput`, including `Pda`, and outputs the `PublicKey`.

The code should work as is though, so feel free to disregard the suggestion.

```suggestion
    const programId = input.compressionProgram
    ? publicKey(input.compressionProgram)
    : context.programs.getPublicKey(
        'splAccountCompression',
        SPL_ACCOUNT_COMPRESSION_PROGRAM_ID
      );
```

Reply:
yes I believe it was to solve the Pda vs Pubkey thing.  Will use your suggestion ty!